e2e/test/scenarios/visualizations-tabular/drillthroughs/column_extract_drill.cy.spec.js

```diff
@@ -0,0 +1,157 @@
+  {
+    option: "Day of month",
+    value: "11",
+  },
+  {
+    option: "Day of week",
+    value: "Tuesday",
+  },
+  {
+    option: "Month of year",
+    value: "Feb",
+  },
+  {
+    option: "Quarter of year",
+    value: "Q1",
+  },
+  {
+    option: "Year",
+    value: "2,025",
+  },
+];
+
+const DATE_QUESTION = {
+  query: {
+    "source-table": ORDERS_ID,
+    aggregation: [
+      ["min", ["field", ORDERS.CREATED_AT, { "base-type": "type/DateTime" }]],
+    ],
+    breakout: [
+      [
+        "field",
+        ORDERS.CREATED_AT,
+        { "base-type": "type/DateTime", "temporal-unit": "month" },
+      ],
+    ],
+    limit: 1,
+  },
+};
+
+describe("extract action", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+    cy.intercept("POST", "/api/dataset").as("dataset");
+    cy.intercept("POST", "/api/card/*/query").as("cardQuery");
+  });
+
+  describe("date columns", () => {
+    describe("should add a date expression for each option", () => {
+      DATE_CASES.forEach(({ option, value }) => {
+        it(option, () => {
+          openOrdersTable({ limit: 1 });
+          cy.wait("@dataset");
+          extractColumnAndCheck({
+            column: "Created At",
+            option,
+            value,
+          });
+        });
+      });
+    });
+
+    it("should add an expression based on a breakout column", () => {
+      cy.createQuestion(DATE_QUESTION, { visitQuestion: true });
+      extractColumnAndCheck({
+        column: "Created At: Month",
+        option: "Month of year",
+        value: "Apr",
+      });
+    });
+
+    it("should add an expression based on an aggregation column", () => {
+      cy.createQuestion(DATE_QUESTION, { visitQuestion: true });
+      extractColumnAndCheck({
+        column: "Min of Created At: Default",
+        option: "Year",
+        value: "2,022",
+      });
+    });
+
+    it("should handle duplicate expression names", () => {
+      openOrdersTable({ limit: 1 });
+      cy.wait("@dataset");
+      extractColumnAndCheck({
+        column: "Created At",
+        option: "Hour of day",
+        newColumn: "Hour of day",
+      });
+      extractColumnAndCheck({
+        column: "Created At",
+        option: "Hour of day",
+        newColumn: "Hour of day_2",
+      });
+    });
+
+    it("should be able to modify the expression in the notebook editor", () => {
+      openOrdersTable({ limit: 1 });
+      cy.wait("@dataset");
+      extractColumnAndCheck({
+        column: "Created At",
+        option: "Year",
+        value: "2,025",
+      });
+      cy.wait("@dataset");
+      openNotebook();
+      getNotebookStep("expression").findByText("Year").click();
+      enterCustomColumnDetails({ formula: "+ 2" });
+      popover().button("Update").click();
+      visualize();
+      cy.wait("@dataset");
+      cy.findByRole("gridcell", { name: "2,027" }).should("be.visible");
+    });
+
+    it("should use current user locale for string expressions", () => {
```

Comment:
I think we should also check "Quarter" translations, it's something that could easily get overlooked.
For example in Polish and Russian quarters should be formatted as Roman numerals.

Reply:
Apparently quarter names are always not localized neither on the FE or BE `(t/format "'Q'Q" t)`